### PR TITLE
chore: Create a prebuild script to generate next data.

### PR DESCRIPTION
**Pull request by Harkunwar**
## Description

Currently we're generating next-data at the lint stage, which isn't the right place. This PR moves that generation to a `prebuild` stage.

## Related Issues
Fixes #5455

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [x] I have run `npx turbo lint` to ensure the code follows the style guide. And run `npx turbo lint:fix` to fix the style errors if necessary.
- [x] I have run `npx turbo format` to ensure the code follows the style guide.
- [x] I have run `npx turbo test` to check if all tests are passing, and/or `npx turbo test:snapshot` to update snapshots if I created and/or updated React Components.
- [x] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

File: scripts/prebuild/generateNextData.mjs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+'use strict';
+
+import * as nextData from '../../next-data/index.mjs';
+
+// generate the node.js releases json file
+await nextData.generateNodeReleasesJson();
+
+// generate the data from blog posts
+await nextData.generateBlogPostsData();
```
Comment (ovflowd) [inline scripts/prebuild/generateNextData.mjs]: Missing EOF

**Review by ovflowd — APPROVED** [review]
LGTM!

**Comment by Harkunwar** [comment]
Merged this PR as it is simple enough and unblocks another PR.

**Comment by ovflowd** [comment]
Just to double-check @Harkunwar you tested this command indeed gets invoked before build, right?

**Comment by Harkunwar** [comment]
@ovflowd Yes, I did check and it was working. 

I just also realized and `prebuild` is a special script keyword and is executed automatically when we run `npm run build` or `npx turbo build`. 

So now it's executing twice since it's also part of the build script. Should we change the name of the script from `prebuild` to something else such as `build:nextData` so there's no uncertainty that `prebuild` will always execute before `build`, or should we just remove it from the `build` script, since it automatically executes?

**Comment by ovflowd** [comment]
I know that prebuild is a special keyword. Any script that starts with “pre” or “post” and as a suffix of another script name, will then either run before or after.

My intent here was to check if the script is generating the files as expected 👀

**Comment by ovflowd** [comment]
I don’t get it why it would be executing twice. Where is it being invoked for the 2nd time?

**Comment by Harkunwar** [comment]
@ovflowd Yes, the files are being generated.

It's being invoked twice because of this change. `build` is calling `prebuild`. I'll make a PR to remove this redundant call.

```json
"build": "cross-env NODE_NO_WARNINGS=1 npm run prebuild && next build",
```

**Pull request merged**